cirq/google/op_deserializer.py

```diff
@@ -158,3 +196,86 @@ def _args_from_proto(
             if value is not None:
                 return_args[arg.constructor_arg_name] = value
         return return_args
+
+
+class CircuitOpDeserializer(OpDeserializer):
+    """Describes how to serialize CircuitOperations."""
+
+    @property
+    def serialized_id(self):
+        return 'circuit'
+
+    def from_proto(
+        self,
+        proto: v2.program_pb2.CircuitOperation,
+        *,
+        arg_function_language: str = '',
+        constants: List[v2.program_pb2.Constant] = None,
+        raw_constants: List[Any] = None,
+    ) -> 'cirq.CircuitOperation':
+        """Turns a cirq.google.api.v2.CircuitOperation proto into a CircuitOperation."""
+        if constants is None or raw_constants is None:
+            raise ValueError(
+                'CircuitOp deserialization requires a constants list and a corresponding list of '
+                'post-deserialization values (raw_constants).'
+            )
+        circuit = raw_constants[proto.circuit_constant_index]
+        if not isinstance(circuit, circuits.FrozenCircuit):
+            raise ValueError(
+                f'Constant at index {proto.circuit_constant_index} was expected to be a circuit, '
+                f'but it has type {type(circuit)} in the raw_constants list.'
+            )
+
+        which_rep_spec = proto.repetition_specification.WhichOneof('repetition_value')
+        if which_rep_spec == "repetition_count":
```

Comment:
We should probably stay consistent and use single quotes.

Reply:
Done, here and below.